ksql-api/src/test/java/io/confluent/ksql/api/BufferedPublisherTest.java

```diff
@@ -0,0 +1,430 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.api;
+
+
+import static io.confluent.ksql.test.util.AssertEventually.assertThatEventually;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.hamcrest.Matchers.nullValue;
+import static org.hamcrest.core.IsInstanceOf.instanceOf;
+
+import io.confluent.ksql.api.TestUtils.AsyncAssert;
+import io.confluent.ksql.api.server.BufferedPublisher;
+import io.vertx.core.Context;
+import io.vertx.core.Vertx;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+
+/**
+ * More BufferedPublisher testing occurs in the TCK tests
+ */
+public class BufferedPublisherTest {
+
+  private Vertx vertx;
+  private Context context;
+  private BufferedPublisher<String> publisher;
+
+  @Before
+  public void setUp() {
+    vertx = Vertx.vertx();
+    context = vertx.getOrCreateContext();
+    publisher = new BufferedPublisher<>(context);
+  }
+
+  @After
+  public void tearDown() {
+    vertx.close();
+  }
+
+  @Test
+  public void shouldCallOnSubscribe() throws Exception {
+    TestSubscriber subscriber = new TestSubscriber(context);
+    subscribeOnContext(subscriber);
+    assertThat(subscriber.getSub(), is(notNullValue()));
+  }
+
+  @Test
+  public void shouldDeliverOneRecordWhenOneIsRequested() throws Exception {
```

Comment:
it seems like most if not all the tests in this class can follow `Given/When/Then`. For these one's it's trivial:
```java
// When
loadPublisher(10);

// Then:
shouldDeliver(1,1);
```

Reply:
Isn't that kind of obvious without the given/when stuff? ;)